Feature 1349 update client documentation (#1982)
mercedes-benz/sechub: sechub-doc/src/docs/asciidoc/documents/code2doc/client/02_sechub_client.adoc

===== scan

Does a synchronous scan which means

 - create a `{sechub}` job
 - upload necessary stuff to `{sechub}` (e.g. zipped sourcefiles for source scans)
 - approve job as ready to start
 - wait until the job is done,
 - download the report file

>  - exits non-zero on `red` (critical or high findings) so your build will break (depends on your build tool)

Reviewer: ```suggestion
 - exits with a non-zero status on `red` (critical or high findings) so your build will break (depends on your build tool)
```

Author: changed

This is the recommended scan action: You only have to make one single client call to trigger a scan and get the report.

IMPORTANT: For security reasons always pass your API-token via the environment variable `SECHUB_APITOKEN`.

**Minimum call syntax**

Prerequisite is a configuration file `sechub.json` in current directory. +

> See here for details: <<section-client-configuration-file,Configuration file>> changeme!!!

Reviewer: What does "changeme!!!" mean?

Author: :smile: That was a marker in case the link reference would be different after editing.
In fact ist has not changed.
-> removed the comment

TIP: The is the most preferred scan action for builds - you only have to make one
     single client call and your are done.
IMPORTANT: For security reasons always pass your API-token via the environment variable `SECHUB_APITOKEN`.

**Minimum call syntax**

Prerequisite is a configuration file `sechub.json` in current directory. +
See here for details: <<section-client-configuration-file,Configuration file>>
See here for details: <<section-client-configuration-file,Configuration file>> changeme!!!
----
SECHUB_APITOKEN=${apitoken}
sechub scan
----

TIP: If you prefer the report in HTML format: +
     `sechub -reportformat html scan`

TIP: With `-output ${file-or-directory}` you can define where to place the report. +
     Can be a directory, a file name or a file path. +
     Default: `sechub_report_${your-project}_${jobUUID}.{json|html|spdx}` in current directory

===== scanAsync

Does scan asynchronous which means
Does scan asynchronous:

  - will create a `{sechub}` job
  - upload necessary stuff to `{sechub}` (e.g. zipped sourcefiles for source scans)
  - approve job as being ready to start

>   - returns `{sechub}` job UUID +

Author: done

===== getStatus
Will fetch the job status and return result as json

**Minimum call syntax**
----
SECHUB_APITOKEN=${apitoken}
sechub getStatus
----

TIP: Defaults to latest job. +

>      You can select the job by providing the wanted job UUID by providing `-jobUUID ${jobUUID}`

Reviewer: ```suggestion
     You can select the job by providing the wanted job UUID by adding `-jobUUID ${jobUUID}`
```

Author: :+1:  That's better.

===== getReport
Will fetch the report as json. (result will only exist when job is done)

**Minimum call syntax**
----
SECHUB_APITOKEN=${apitoken}
sechub getReport
----

TIP: Defaults to latest finished job. +

>      You can select the job by providing the wanted job UUID by providing `-jobUUID ${jobUUID}`

Reviewer: ```suggestion
     You can select the job by providing the wanted job UUID by adding `-jobUUID ${jobUUID}`
```

TIP: With `-output ${file-or-directory}` you can define where to place the report. +
     Can be a directory, a file name or a file path. +
     Default: `sechub_report_${your-project}_${jobUUID}.{json|html|spdx}` in current directory

===== listJobs
Lists the last jobs and their state.

**Call syntax**
----
SECHUB_APITOKEN=${apitoken}
sechub listJobs
----

> TIP: This might be valuable for an overview +

Author: done

> - `s` Skip following findings - End the asking here and add the marked items to the false positives list on the {sechub} server.

Reviewer: ```suggestion
- `s` Skip following findings - Finish the marking and add the marked items to the false positives list on the {sechub} server.
```

Author: done

`export http_proxy=http://localhost:3128` (e.g. when using a local CNTLM proxy)

==== Debugging

> For error hunting, you can

Reviewer: ```suggestion
For troubleshooting you can
```

Author: done
